src/gallery/actions/delete-gallery.action.ts

```diff
@@ -9,13 +10,31 @@ export const deleteGallery = async (
   galleryId: string,
 ): Promise<IGallery> => {
   try {
-    const response = await fetcher.delete<IGallery>(
-      StringUtils.bindContext(endpoints.GALLERY, {
-        id: galleryId,
-      }),
+    const response = await fetcher
+      .delete<IGallery>(
+        StringUtils.bindContext(endpoints.GALLERY, {
+          id: galleryId,
+        }),
+      )
+      .catch(() => {
+        throw errorFactory.create(GalleryErr.DELETE_GALLERY_FORBIDDEN);
+      });
+
+    switch (response.status) {
+      case 200:
+        return response.data;
+      case 403:
+        throw errorFactory.create(GalleryErr.DELETE_GALLERY_FORBIDDEN);
+      case 404:
+        throw errorFactory.create(GalleryErr.DELETE_GALLERY_NOT_FOUND);
```

Comment:
En suivant mon autre commentaire concernant le code d'erreur, ici on aurait : 

```suggestion
        throw errorFactory.create(GalleryErr.GALLERY_NOT_FOUND);
```